galata/src/benchmarkReporter.ts

```diff
@@ -178,17 +178,35 @@ class BenchmarkReporter implements Reporter {
    *   comparison: Logic of test comparisons: 'snapshot' or 'project'
    *    - 'snapshot': (default) This will compare the 'actual' result with the 'expected' one
    *    - 'project': This will compare the different project
+   *   graphConfigFactory: Function to create  VegaLite configuration from test records.
+   *   mdTableFactory: Function to create  markdown table from test records.
    */
   constructor(
-    options: { outputFile?: string; comparison?: 'snapshot' | 'project' } = {}
+    options: {
+      outputFile?: string;
+      comparison?: 'snapshot' | 'project';
+      graphConfigFactory?: (
+        allData: Array<IReportRecord>,
+        comparison: 'snapshot' | 'project'
+      ) => Record<string, any>;
+      mdTableFactory?: (allData: Array<IReportRecord>) => Array<string>;
```

Comment:
As the result of this factory is directly dumped in a file, it will be nice to change the signature like

```suggestion
      textReportFactory?: (allData: Array<IReportRecord>) => Promise<[string, string]>;
```

The `Promise` is to allow downstream usage of third party library that may be async. And the idea of returning two strings is too returned the (`content`, `extension`) to allow other format than Markdown.